pkg/controller/plan/kubevirt.go

```diff
@@ -503,12 +523,69 @@ func (r *KubeVirt) EnsureDataVolumes(vm *plan.VMStatus, dataVolumes []cdi.DataVo
 					dv.Name),
 				"vm",
 				vm.String())
+			vm.VolumeNames = append(vm.VolumeNames, dv.Name)
 		}
 	}
 
 	return
 }
 
+//
+// Return DataVolumes associated with a VM.
+func (r *KubeVirt) getDVs(vm *plan.VMStatus) (dvs []DataVolume, err error) {
+	dvsList := &cdi.DataVolumeList{}
+	err = r.Destination.Client.List(
+		context.TODO(),
+		dvsList,
+		&client.ListOptions{
+			LabelSelector: labels.SelectorFromSet(r.vmLabels(vm.Ref)),
+			Namespace:     r.Plan.Spec.TargetNamespace,
+		})
+
+	if err != nil {
+		err = liberr.Wrap(err)
+		return
+	}
+
+	dvs = []DataVolume{}
+	for i := range dvsList.Items {
+		dv := &dvsList.Items[i]
+		dvs = append(dvs, DataVolume{
+			DataVolume: dv,
+			PVC:        nil,
+		})
+	}
+	return
+}
+
+//
+// Return PersistentVolumeClaims associated with a VM.
+func (r *KubeVirt) getPVCs(vm *plan.VMStatus) (pvcs []core.PersistentVolumeClaim, err error) {
+	pvcsList := &core.PersistentVolumeClaimList{}
+	err = r.Destination.Client.List(
+		context.TODO(),
+		pvcsList,
+		&client.ListOptions{
+			Namespace: r.Plan.Spec.TargetNamespace,
+		})
+
+	if err != nil {
+		err = liberr.Wrap(err)
+		return
+	}
```

Comment:
this part seen here and there, refactoring might be good, at least to the list options.